plugins/regrass.cpp

```diff
@@ -12,109 +27,553 @@ using namespace DFHack;
+
+    static struct_identity _identity;
+};
+static const struct_field_info regrass_options_fields[] =
+{
+    { struct_field_info::PRIMITIVE, "max_grass",    offsetof(regrass_options, max_grass),    &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "new_grass",    offsetof(regrass_options, new_grass),    &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "force",        offsetof(regrass_options, force),        &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "ashes",        offsetof(regrass_options, ashes),        &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "buildings",    offsetof(regrass_options, buildings),    &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "mud",          offsetof(regrass_options, mud),          &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "block",        offsetof(regrass_options, block),        &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "zlevel",       offsetof(regrass_options, zlevel),       &df::identity_traits<bool>::identity, 0, 0 },
+    { struct_field_info::PRIMITIVE, "forced_plant", offsetof(regrass_options, forced_plant), &df::identity_traits<int32_t>::identity, 0, 0 },
+    { struct_field_info::END }
+};
+struct_identity regrass_options::_identity(sizeof(regrass_options), &df::allocator_fn<regrass_options>, NULL, "regrass_options", NULL, regrass_options_fields);
+
+command_result df_regrass(color_ostream& out, vector<string> &parameters);
+
+static bool valid_tile(regrass_options options, df::map_block *block, int x, int y)
+{   // Is valid tile for regrass
+    auto des = block->designation[x][y];
+    auto tt = block->tiletype[x][y];
+    auto shape = tileShape(tt);
+    auto mat = tileMaterial(tt);
+    auto spec = tileSpecial(tt);
+
+    if (mat == tiletype_material::GRASS_LIGHT ||
+        mat == tiletype_material::GRASS_DARK ||
+        mat == tiletype_material::PLANT) // Shrubs and saplings can have grass underneath
+    {   // Refill existing grass
+        DEBUG(log).print("Valid tile: Grass/Shrub/Sapling\n");
+        return true;
+    }
+    else if (tt == tiletype::TreeTrunkPillar)
+    {   // Trees can have grass under main tile
+        auto p = df::coord(block->map_pos.x + x, block->map_pos.y + y, block->map_pos.z);
+        auto mbc = world->map.column_index[(block->map_pos.x / 48)*3][(block->map_pos.y / 48)*3];
+        if (!mbc)
+        {
+            Core::printerr("No MLT block column for tile at (%d, %d, %d)!\n", p.x, p.y, p.z);
+            return false;
+        }
+
+        for (auto plant : mbc->plants)
+        {
+            if (plant->pos == p)
+            {
+                DEBUG(log).print("Valid tile: Tree\n");
+                return true; // Is main tile
+            }
+        }
```

Comment:
Yeah. Just realized I'm going to have to check adjacent tiles for thick trees, though, so I'm going to have to use the original version anyway.

Reply:
Doesn't look like `getPlantAtTile` is really used anywhere else. Do you still want the extra option?

Reply:
Yeah, it should keep the existing behavior by default. It's used by `Gui::getAnyPlant` for getting the plant under the keyboard cursor

Reply:
`getAnyPlant` is going to use the branch version, and `regrass` is going to use the branch version (because of 2x2 trunks.)

There isn't anywhere that would currently use the skip branch version.

Reply:
Got it. Yeah if everyone is using the same version, no need to parameterize